doc/embedded/crosscompile_example.md

```diff
@@ -0,0 +1,194 @@
+## Crosscompile mlpack example for an embedded hardware
+
+In this article, we explore how to crosscompile and run an mlpack example code
+on an embedded hardware. In our previous
+[documentations](crosscompile_armv7.cmake), we have explored how to run mlpack
+bindings such as kNN command line program on a Raspberry PI 2. Please refer to that
+article first and follow the first part on how to setup cross-compilation toolchain,
+and then continue with this article. Or refer to this
+[table](supported_boards.md) for a quick start guide.
+
+### Cloning mlpack example respository
+
+mlpack has an [example repository](https://github.com/mlpack/examples) that
+demonstrates a set of examples showing how to use the library source code
+on different dataset and usecases including embedded deployment. This
+tutorial basically explains our necessary CMake configurations
+that are required to integrate with your local CMake to download mlpack
+dependencies and cross compile the entire software.
```

Comment:
```suggestion
shows a number of applications and use cases for mlpack, including embedded
deployment.  In this tutorial, we are interested in the `embedded/` directory,
which provides a CMake project template that compiles a random forest
application to embedded hardware.  This project template can be adapted to a new
project, or its pieces can be incorporated into an existing CMake project.
```

I tried to shorten this a little bit---we can remove the paragraph below (for the most part) too as a result.  Let me know what you think.